hail/python/hail/methods/impex.py

```diff
@@ -1891,9 +1895,81 @@ def import_vcf(path,
 
     reader = MatrixVCFReader(path, call_fields, header_file, min_partitions,
                              reference_genome, contig_recoding, array_elements_required,
-                             skip_invalid_loci, force_bgz, force)
+                             skip_invalid_loci, force_bgz, force, _partitions)
     return MatrixTable(MatrixRead(reader, drop_cols=drop_samples))
 
+@typecheck(path=sequenceof(str),
+           partitions=str,
+           force=bool,
+           force_bgz=bool,
+           call_fields=oneof(str, sequenceof(str)),
+           reference_genome=nullable(reference_genome_type),
+           contig_recoding=nullable(dictof(str, str)),
+           array_elements_required=bool,
+           skip_invalid_loci=bool)
+def import_vcfs(path,
+                partitions,
+                force=False,
+                force_bgz=False,
+                call_fields=[],
+                reference_genome='default',
+                contig_recoding=None,
+                array_elements_required=True,
+                skip_invalid_loci=False) -> MatrixTable:
+    """Experimental. Import multiple vcfs as :class:`MatrixTable`s
+
+    The arguments to this function are almost identical to :func:`.import_vcf`,
+    the only difference is the `partitions` argument, which is used to divide
+    and filter the vcfs. It must be a JSON string that will deserialize to an
+    Array of Intervals of Locus structs. A partition will be created for every
+    element of the array. Loci that fall outside of any interval will not be
+    imported. For example:
+
+    .. code-block:: text
+        [
+          {
+            "start": {
+              "locus": {
+                "contig": "chr22",
+                "position": 1
+              }
+            },
+            "end": {
+              "locus": {
+                "contig": "chr22",
+                "position": 5332423
+              }
+            },
+            "includeStart": true,
+            "includeEnd": true
+          }
+        ]
+
+    The `includeStart` and `includeEnd` keys must be `true`. The `contig` fields must
+    be the same.
+
+    One difference between :func:`.import_vcfs` and :func:`.import_vcf` is that
+    :func:`.import_vcfs` only keys the resulting matrix tables by `locus`
+    rather than `locus, alleles`.
+    """
+
+    rg = reference_genome.name if reference_genome else None
+
+    global _cached_importvcfs
+    if _cached_importvcfs is None:
+        _cached_importvcfs = Env.hail().io.vcf.ImportVCFs
+
+    jmts = _cached_importvcfs.pyApply(
```

Comment:
Is there an easy way for this to go through the existing IR infrastructure `(MatrixRead (reader) drop_cols drop_rows)`? If not, I'll make a separate PR eventually that does that.

Reply:
I don't think it's easy, but I don't think it's hard either. It probably requires adding some new `MatrixReader` to handle the differences between `ImportVCF` and `ImportVCFs` and all the infrastructure that comes along with adding a new IR. Again, not difficult but I'm not sure I can spend the time to handle it right now.